do-server/src/index.mjs

```diff
@@ -1,19 +1,36 @@
 import { mutators } from '../../frontend/src/utils/mutators.js';
 
 class ServerTransaction {
-  constructor(canon, transactionID, mutatorArgs) {
+  constructor(canon, transactionID, mutator, mutatorArgs, patch) {
     this.transactionID = transactionID;
+    this.mutator = mutator;
     this.mutatorArgs = mutatorArgs;
     this.canon = canon;
+    this.patch = patch;
   }
 
   get(key) {
     return this.canon[key];
   }
 
   set(key, value) {
+    this.patch.push({
+      op: 'put',
+      key,
+      value,
+    });
+
     this.canon[key] = value;
```

Comment:
Could this potentially go above pushing the object into the patch array? Same goes for the delete method below.

Reply:
Good call! We probably want to make sure the values get changed in the sever state before putting them in the patch array just to ensure the client never gets ahead of the server. Will fix now.

Reply:
new commit pushed